pkg/cmd/pulumi/pulumi.go

```diff
@@ -397,6 +398,38 @@ func NewPulumiCmd() *cobra.Command {
 	return cmd
 }
 
+// haveNewerDevVersion checks whethere we have a newer dev version available.
+func haveNewerDevVersion(devVersion semver.Version, curVersion semver.Version) bool {
```

Comment:
This function is intentionally lenient with any errors, erring on the side of pretending we don't have a newer version available.  I expect this to be a better user experience than adding any asserts here for correctness, as this affects merely a warning, and if we don't end up showing it it should not be a big problem.

Reply:
Good idea, done :heavy_check_mark: